### ISSUE TITLE
Fine-tuning Reproducibility

### ISSUE THREAD

**Issue by dongmean** [allenai/open-instruct#66]
Hi, first of all, thanks a lot for your great contributions!

I'm trying to fine-tune llama-7b on alpaca-52k, but facing a performance reproducibility issue for the MMLU benchmark.
I used finetune_with_accelerate.sh for two A100 40gb with the same hyperparameters in the paper as below

```
export CUDA_VISIBLE_DEVICES=0,1

MODEL_SIZE=7B
NUM_GPUS=2
BATCH_SIZE_PER_GPU=2
TOTAL_BATCH_SIZE=128
GRADIENT_ACC_STEPS=$(($TOTAL_BATCH_SIZE/$NUM_GPUS/$BATCH_SIZE_PER_GPU))
echo "Training llama model ${MODEL_SIZE} using $NUM_GPUS GPUs, $BATCH_SIZE_PER_GPU batch size per GPU, $GRADIENT_ACC_STEPS gradient accumulation steps"

accelerate launch \
    --mixed_precision bf16 \
    --num_machines 1 \
    --num_processes $NUM_GPUS \
    --use_deepspeed \
    --deepspeed_config_file ds_configs/stage3_offloading_accelerate.conf \
    open_instruct/finetune.py \
    --model_name_or_path decapoda-research/llama-7b-hf \
    --tokenizer_name decapoda-research/llama-7b-hf \
    --use_slow_tokenizer \
    --train_file /datasets/open_instruct/stanford_alpaca/stanford_alpaca_data.jsonl\
    --use_flash_attn \
    --max_seq_length 2048 \
    --preprocessing_num_workers 16 \
    --per_device_train_batch_size $BATCH_SIZE_PER_GPU \
    --gradient_accumulation_steps $GRADIENT_ACC_STEPS \
    --learning_rate 2e-5 \
    --lr_scheduler_type linear \
    --warmup_ratio 0.03 \
    --weight_decay 0. \
    --num_train_epochs 2 \
    --output_dir /checkpoints/llama-7b-dm-alpaca52k \
    --with_tracking \
    --report_to tensorboard \
    --logging_steps 1
```

Below is the script for MMLU evaluation

```
export CUDA_VISIBLE_DEVICES=1

# Evaluating llama 7B model using 0 shot directly

python -m eval.mmlu.run_eval \
    --ntrain 0 \
    --data_dir /datasets/MMLU/data \
    --save_dir results/mmlu/llama-7B-0shot \
    --model_name_or_path /checkpoints/llama-7b-dm-alpaca52k \
    --tokenizer_name_or_path /checkpoints/llama-7b-dm-alpaca52k \
    --eval_batch_size 4 \
    --load_in_8bit \
    --use_chat_format
```

However, I got an average accuracy of 0.320 which was 0.415 in the paper.

I already checked 0.415 was reproducible when I downloaded the officially released model at [allenai/open-instruct-stanford-alpaca-7b](https://huggingface.co/allenai/open-instruct-stanford-alpaca-7b).
So, the performance drop would be from fine-tuning.

Can anyone give me any tips for what I might miss?
Thanks

**Comment by yizhongw** [comment]
Hi @dongmean thanks for reporting this. Could you check your pip installed packages are consistent with our `requirements.txt`? Especially, the latest accelerate has some issue so I have to downgrade it. See here: https://github.com/allenai/open-instruct/commit/662c2ad497d43054b219f4bf52820a5d7133e359

**Comment by dongmean** [comment]
Hi @yizhongw, thanks for the reply. Following your suggestion, I aligned my pip packages with the requirements.txt and downgraded the accelerate package, but still have the same issue. Is there some possibility that I should use some specific optimization configuration for alpaca-52k? found out that the default opt configuration is different from the one in the original alpaca-52k paper. Thanks.

**Comment by lucywang720** [comment]
hello! I am wondering how long it takes to train your tulu?

**Comment by hamishivi** [comment]
Hi - training time can take roughly a day or two on a 8 x 80GB A100 node for the 7B model, and longer for larger models. Closing issue due to inactivity, but feel free to re-open if you are still having issues.

One large thing that can change is the amount of gradient accumulation you use - you should try to use as little gradient accumulation as possible, since this changes how examples are weighted.
